packages/react-router/index.tsx

```diff
@@ -543,6 +541,13 @@ function useRoutes_(
     ? joinPaths([parentPathname, basename])
     : parentPathname;
 
+  let contextLocation = useLocation();
+  let location = React.useMemo(() => {
+    return locationOverride
+      ? { ...contextLocation, ...locationOverride }
```

Comment:
Why merge the two locations? I feel like this could cause subtle bugs.

Reply:
Hmm, any examples you can think of? The rationale here is that a user may provide a partial location object, which I believe we should probably support.

Reply:
Well in the future if the Location object changes (like you add a field to it) then I'd rather my typescript compiler complain than implicitly pulling this new field from the context.